qcengine/programs/molpro.py

```diff
@@ -293,32 +311,39 @@ def parse_output(self, outfiles: Dict[str, str], input_model: 'ResultInput') ->
         mol_method = molecule.attrib['method']
         mol_final_energy = float(molecule.attrib['energy'])
 
+        # Grab the method from input
+        method = input_model.model.method
+        if method.upper() in self._dft_functionals:  # Determine if method is a DFT functional
+            method = "RKS"
+
         # A slightly more robust way of determining the final energy.
         # Throws an error if the energy isn't found for the method specified from the input_model.
-        # TODO Will need to be modified to work for DFT. XC --> RKS
-        method = input_model.model.method
-        method_energy_map = supported_methods[method]['energy']
-        if method in post_hf_methods and method_energy_map['total energy'] in properties:
-            final_energy = properties[method_energy_map['total energy']]
-        elif method in scf_methods and method_energy_map['Energy'] in properties:
-            final_energy = properties[method_energy_map['Energy']]
+        if method in self._post_hf_methods and molpro_map['total energy'][method] in properties:
+            final_energy = properties[molpro_map['total energy'][method]]
+        elif method in self._scf_methods and molpro_map['Energy'][method] in properties:
+            final_energy = properties[molpro_map['Energy'][method]]
         else:
+            # TODO This won't work for DFT case since mol_method will be name of DFT functional
```

Comment:
Which cases will not work here and what is the data that will be missing?

Reply:
So this is meant to serve as a backup way of grabbing the energy if for some reason it’s not provided in the xml normally. This happens in the case of calling a CCSD or a CCSD(T) gradient. So this shouldn’t be needed for a DFT calculation but right now it wouldn’t work since mol_method is the actual functional name and method at this point is RKS.

Reply:
I just made a fix for this through some logic rearrangement. Will push now.